### PR TITLE
Update initialize-atom-on-render.mdx

### PR DESCRIPTION
**Pull request by ioExpander**
## Related Issues or Discussions
https://github.com/pmndrs/jotai/discussions/2262

Fixes #

## Summary
Added a Typescript example for useHydrateAtoms

## Check List

- [ ] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: docs/guides/initialize-atom-on-render.mdx [hunk 1]
```diff
@@ -93,3 +93,46 @@ This behavior is due to our child components looking for the lowest commmon `Pro
 For more information on `Provider` behavior, please read the docs [here](../core/provider.mdx).
 
 For more complex use cases, check out [Scope extension](../extensions/scope.mdx).
+
+
+### Using Typescript
+
+`useHydrateAtoms` has overloaded types and typescript cannot extract types from overloaded function. It is recommended to use a `Map` when passing initial atom values to the `useHydrateAtoms`.
+
+Here is a working example :
```
Comment (dai-shi) [inline docs/guides/initialize-atom-on-render.mdx:102]:
```suggestion
Here is a working example:
```

File: docs/guides/initialize-atom-on-render.mdx [hunk 2]
```diff
@@ -93,3 +93,46 @@ This behavior is due to our child components looking for the lowest commmon `Pro
...
+
+```jsx
```
Comment (dai-shi) [inline docs/guides/initialize-atom-on-render.mdx:104]:
```suggestion
```tsx
```

File: docs/guides/initialize-atom-on-render.mdx [hunk 3]
```diff
@@ -93,3 +93,46 @@ This behavior is due to our child components looking for the lowest commmon `Pro
...
+import { Provider, atom, useAtomValue, type WritableAtom } from "jotai";
+import { useHydrateAtoms } from "jotai/utils";
+import { type ReactNode } from "react";
```
Comment (dai-shi) [inline docs/guides/initialize-atom-on-render.mdx:107]: Just a preference.
```suggestion
import type { ReactNode } from "react";
import { Provider, atom, useAtomValue } from "jotai";
import type { WritableAtom } from "jotai";
import { useHydrateAtoms } from "jotai/utils";
```

**Review by dai-shi — COMMENTED** [review]
Thanks for improving docs! Can you run prettier?

File: docs/guides/initialize-atom-on-render.mdx (hunk 2 above)
Comment (himself65) [reply docs/guides/initialize-atom-on-render.mdx:104]: No, I think it's jsx, all other documents are jsx

File: docs/guides/initialize-atom-on-render.mdx [hunk 4]
```diff
@@ -93,3 +93,46 @@ This behavior is due to our child components looking for the lowest commmon `Pro
...
+
+const testAtom = atom("");
+
+export default function JotaiTests() {
```
Comment (himself65) [inline docs/guides/initialize-atom-on-render.mdx:111]: Naming looks weird.

How about, App, Component, AtomsHydrator

**Comment by dai-shi** [comment]
@ioExpander Are you still around?

**Comment by ioExpander** [comment]
> @ioExpander Are you still around?

Ah yes ! I got distracted by work related stuff and had completely forgotten about this. 🫢 Thank you for the reminder.
Ran prettier and resolved comments this morning.

**Review by dai-shi — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**